Source/Private/common.ps1

```diff
@@ -375,18 +375,27 @@ function _buildRequestURI {
          $sb.Append("/$id") | Out-Null
       }
 
+      $QSParams = [System.Collections.ArrayList]@()
       if ($version) {
-         $sb.Append("?api-version=$version") | Out-Null
+         $QSParams.Add("api-version=$version") | Out-Null
       }
 
-      $url = $sb.ToString()
-
       if ($queryString) {
          foreach ($key in $queryString.keys) {
-            $Url += _appendQueryString -name $key -value $queryString[$key]
+            $value = $queryString[$key]
+            if($value) {
+               $QSParams.Add("$key=$value") | Out-Null
+            }
          }
       }
 
+      if($QSParams.Count) {
+         $sb.Append("?") | Out-Null
+         $sb.Append($QSParams -join "&") | Out-Null
+      }
+
+      $url = $sb.ToString()
```

Comment:
Why not using the _appendQueryString and refactor that function?

Somehow it feels like the fix needs a bit more refactoring on also different places to make the code cleaner. Because now we have an approach here to deal with the query string and another with the function (which is not the same as your code but potion of it is trying to do the same).

Reply:
Having a string builder for initial portion of the URL, then growing it by `+=` anyway, is weird to begin with.

Wrapping a `($key, $value) => "$key=$value"` logic into a separate function doesn't make sense to me either :) If it was my project, I'd use a built-in query string generator - .NET RTL has several. For compatibility with .NET 4.x, I'd use `HttpValueCollection`, as constructed via `HttpUtility.Parse()`. It's in the `System.Web.HttpUtility` assembly though, requires an `Add-Type` call.

What do you think? You are the maintainer, the coding style is your decision after all.

Reply:
I am ok with this. Would you give it a take to use a build it .net query generator for another PR? I would be fine with it if it improves the things. But I would need unit tests for the wrapping function, and it should be replaced / tested everywhere, where it is used.